src/components/SingleQuestion.tsx

```diff
@@ -60,29 +50,38 @@ const SingleQuestion = ({
 
   return (
     <>
-      <div>
+      <div className="question-container">
         <h1 className="progress">
           Question <span className="question-number">{currentQNo + 1}</span>/
           {totalQuestions}
         </h1>
-        <p className="question-text">{questionInfo.question}</p>
+        <p className="question-text">{decodeHtml(questionInfo.question)}</p>
       </div>
-      <div className="answer-grid">
+      <div className="answer-grid-container">
         {allAnswers.map((answer, idx) => (
           <button
-            className={
-              selectedAnswer === idx
-                ? "answer-btn selected-answer-btn"
-                : "answer-btn"
-            }
+            className={[
+              "answer-btn",
+              selectedAnswer === idx && "selected-answer-btn",
+              answerIsSubmitted &&
+                selectedAnswer === idx &&
+                allAnswers[selectedAnswer!].isCorrect === false &&
+                "incorrect-answer-btn",
+              answerIsSubmitted &&
+                allAnswers[idx].isCorrect === true &&
+                "correct-answer-btn",
+              answerIsSubmitted && "locked",
+            ]
+              .filter(Boolean)
+              .join(" ")}
```

Comment:
this should be extracted to a helper function,

Reply:
Just to clarify, are you suggesting that I extract the entire button as a separate component?

Reply:
sorry, I meant extract the following into a separate function: ie:
```ts
const getClass = () => {
  const classes = ["answer-btn"];
  const isSelected = selectedAnswer === idx;
  const isCorrect = allAnswers[idx].isCorrect;
  const isSelectedCorrect = selectedAnswer !== null && allAnswers[selectedAnswer]?.isCorrect;

  if (isSelected) {
    classes.push("selected-answer-btn");
  }

  if (answerIsSubmitted) {
    classes.push("locked");

    if (isSelected && !isSelectedCorrect) {
      classes.push("incorrect-answer-btn");
    }

    if (isCorrect) {
      classes.push("correct-answer-btn");
    }
  }

  return classes.join(" ");
};
```